LFS171x/docs/introduction-to-hyperledger-fabric.md

```diff
@@ -1 +1,1547 @@
+
+**Troubleshooting Note:** 
+If you have difficulties with the two previous commands and you suspect that your Docker images may be at fault, you can start back from scratch, which will delete and untag the Docker images.
+
+```
+$ docker rmi -f $(docker images -q)
+```
+
+Once you run this command, return to the _Installing Hyperledger Fabric Docker Images and Binaries_ page, at the beginning of this section.
+
+### Finishing Up and Shutting Down the Network
+
+Finally, let’s test bringing down this network.
+
+Within the same terminal, do **Control+C** to exit the current execution.
+
+Then, run the following command:
+
+```
+$ ./byfn.sh -m down
+```
+
+Another command line will appear, reply with **Y \<Enter>** to continue.
+
+This command will kill your containers, remove the crypto material and four artifacts, and delete the chaincode images from your Docker Registry.
+
+And that’s it for a simple demonstration!
+
+These simple steps show how we can easily spin up and bring down a Hyperledger Fabric network, given the code we have. In the next section, we will learn more about chaincode.
+
+### Video: Installing Hyperledger Fabric (Demo)
+
+#### Installing Hyperledger Fabric (Demo)
+
+[![Installing Hyperledger Fabric (Demo)](../images/video-image.png)](https://youtu.be/eM6HIOpexGE)
+
+## Understanding Chaincode
+
+### Chaincode
+
+In Hyperledger Fabric, **chaincode** is the 'smart contract' that runs on the peers and creates transactions. More broadly, it enables users to create transactions in the Hyperledger Fabric network's shared ledger and update the world state of the assets.
+
+Chaincode is programmable code, written in Go, and instantiated on a channel. Developers use chaincode to develop business contracts, asset definitions, and collectively-managed decentralized applications. The chaincode manages the ledger state through transactions invoked by applications. Assets are created and updated by a specific chaincode, and cannot be accessed by another chaincode.
+
+Applications interact with the blockchain ledger through the chaincode. Therefore, the chaincode needs to be installed on every peer that will endorse a transaction and instantiated on the channel.
+
+There are two ways to develop smart contracts with Hyperledger Fabric:
+
+ - Code individual contracts into standalone instances of chaincode
+ - (More efficient way) Use chaincode to create decentralized applications that manage the lifecycle of one or multiple types of business contracts, and let the end users instantiate instances of contracts within these applications.
+ 
+### Chaincode Key APIs
+
+An important interface that you can use when writing your chaincode is defined by Hyperledger Fabric - <span style="color:cyan">_ChaincodeStub_</span> and <span style="color:cyan">_ChaincodeStubInterface_</span>. The ChaincodeStub provides functions that allow you to interact with the underlying ledger to query, update, and delete assets. The key APIs for chaincode include:
```

Comment:
Remove color spans from here. The _ChaincodeStub_ and _ChaincodeStubInterface_ are actually links.
https://godoc.org/github.com/hyperledger/fabric/core/chaincode/shim#Chaincode
https://godoc.org/github.com/hyperledger/fabric/core/chaincode/shim#ChaincodeStub